lib/kill.js

```diff
@@ -72,8 +72,7 @@ export const waitForSuccessfulExit = async exitPromise => {
 	const [exitCode, signal] = await exitPromise;
 
 	if (!isProcessErrorExit(exitCode, signal) && isFailedExit(exitCode, signal)) {
-		const message = getErrorPrefix(normalizeExitPayload(exitCode, signal));
-		throw setDiscardedError(new Error(message));
```

Comment:
Those `DiscardedError`s used to set some basic error message, because I was thinking this would be used on the upcoming PR related to improving the `.pipe()` method. However, it turns out this is not needed, so this can be removed. It is not used right now.